src/Middleware.php

```diff
@@ -51,8 +51,29 @@ public static function sourceFile(
                 if ($source !== null
                     && $operation->getInput()->hasMember($bodyParameter)
                 ) {
-                    $command[$bodyParameter] = new LazyOpenStream($source, 'r');
+                    $lazyOpenStream = new LazyOpenStream($source, 'r');
+                    $command[$bodyParameter] = $lazyOpenStream;
                     unset($command[$sourceParameter]);
+
+                    $next = $handler($command, $request);
+                    // To avoid failures in some tests cases
+                    if ($next !== null && method_exists($next, 'then')) {
+                        try {
```

Comment:
Should this not be:

```php
if ($next !== null && method_exists($next, 'then')) {
    return $next->then(
        function ($result) use ($lazyOpenStream)
            $lazyOpenStream->close();
            return $result;
        }
    )->otherwise(function (\Throwable $e) use ($lazyOpenStream) {
        $lazyOpenStream->close();
        throw $e; // Re-throw the exception
    });
}
```